skbio/stats/tests/test_composition.py

```diff
@@ -269,6 +355,136 @@ def test_centralize(self):
                             np.array([[2, 2, 6],
                                       [4, 4, 2]]))
 
+    def test_ancom_basic_counts(self):
+        test_table = pd.DataFrame(self.table1)
+        original_table = copy.deepcopy(test_table)
+        result = ancom(test_table,
+                       pd.Series(self.cats1),
+                       multiple_comparisons_correction=None)
+
+        # Test to make sure that the input table hasn't be altered
+        npt.assert_allclose(original_table, test_table)
+        exp = pd.DataFrame({'W': np.array([5, 5, 2, 2, 2, 2, 2]),
+                            'reject': np.array([True, True, False, False,
+                                                False, False, False],
+                                               dtype=bool)})
+        assert_data_frame_almost_equal(result, exp)
+
+    def test_ancom_basic_proportions(self):
+        # Converts from counts to proportions
+        test_table = closure(self.table1)
+        original_table = copy.deepcopy(test_table)
+        result = ancom(test_table,
+                       self.cats1,
+                       multiple_comparisons_correction=None)
+        # Test to make sure that the input table hasn't be altered
+        npt.assert_allclose(original_table, test_table)
```

Comment:
I would write a separate unit test that checks that `table` and `grouping` aren't side-effected by the function (this follows [Arrange Act Assert](http://c2.com/cgi/wiki?ArrangeActAssert)).